cli/subcommands/src/cargo_hax.rs

```diff
@@ -66,25 +73,361 @@ fn rustflags() -> String {
+                }
+                FromEngine::File(file) => {
+                    if backend.dry_run {
+                        output.files.push(file)
+                    } else {
+                        let path = out_dir.join(&file.path);
+                        std::fs::create_dir_all(&path.parent().unwrap()).unwrap();
+                        std::fs::write(&path, file.contents).unwrap();
+                        let title = format!("hax: wrote file {}", path.display());
+                        report(Level::Info.title(&title))
+                    }
+                }
+                FromEngine::DebugString(debug) => {
+                    output.debug_json = Some(debug);
+                }
+                FromEngine::PrettyPrintDiagnostic(diag) => {
+                    send!(&ToEngine::PrettyPrintedDiagnostic(format!("{}", diag)));
+                }
+                FromEngine::PrettyPrintRust(code) => {
+                    let code = match syn::parse_file(&code) {
+                        Ok(file) => Ok(prettyplease::unparse(&file)),
+                        Err(err) => Err(format!("{}", err)),
+                    };
+                    send!(&ToEngine::PrettyPrintedRust(code));
+                }
+                FromEngine::Ping => {
+                    send!(&ToEngine::Pong);
+                }
+            }
+        }
+        drop(stdin);
+    }
+
+    let exit_status = engine_subprocess.wait().unwrap();
+    if !exit_status.success() {
+        let title = format!(
+            "hax: {} exited with non-zero code {}",
+            ENGINE_BINARY_NAME,
+            exit_status.code().unwrap_or(-1),
+        );
+        report(Level::Error.title(&title));
+        std::process::exit(1);
+    }
+
+    if backend.dry_run {
+        serde_json::to_writer(std::io::BufWriter::new(std::io::stdout()), &output).unwrap()
+    }
+    if let Some(debug_json) = &output.debug_json {
+        use DebugEngineMode;
+        match &backend.debug_engine {
+            Some(DebugEngineMode::Interactive) => {
+                eprintln!("----------------------------------------------");
+                eprintln!("----------------------------------------------");
+                eprintln!("----------------------------------------------");
+                eprintln!("-- Engine debug mode. Press CTRL+C to exit. --");
+                eprintln!("----------------------------------------------");
+                eprintln!("----------------------------------------------");
+                eprintln!("----------------------------------------------");
+                hax_phase_debug_webapp::run(|| debug_json.clone())
+            }
+            Some(DebugEngineMode::File(_file)) if !backend.dry_run => {
+                println!("{}", debug_json)
+            }
+            _ => (),
+        }
+    }
+
+    error
+}
+
+/// Uses `cargo metadata` to compute a derived target directory.
+fn target_dir(suffix: &str) -> camino::Utf8PathBuf {
```

Comment:
Do we really need this dependency? Why are general paths not good enough?

Reply:
No, `PathBuf`s are fine, but `cargo_metadata` gives you camino paths. But actually I can just `into` that camino path and kill the (direct) dependency